CMakeLists.txt

```diff
@@ -169,9 +169,35 @@ if(${BLAS} STREQUAL "MKL" OR DEFINED ENV{MKLROOT})
         -lmkl
         CACHE STRING "")
   endif()
-elseif(NOT DEFINED ENV{BLAS_LIBS} AND NOT DEFINED ENV{BLAS_CFLAGS})
+elseif(
+  NOT DEFINED ENV{BLAS_LIBS}
+  AND NOT DEFINED ENV{BLAS_CFLAGS}
+  AND NOT BLAS_FOUND)
   # if nothing is specified via environment variables, let's try FindBLAS
-  find_package(BLAS)
+  if($(CMAKE_VERSION) VERSION_GREATER_EQUAL 3.22)
+    set(BLA_SIZEOF_INTEGER 8)
+  endif()
+
+  if(APPLE AND (NOT DEFINED BLA_VENDOR OR BLA_VENDOR STREQUAL "All"))
```

Comment:
```suggestion
  if(APPLE AND (NOT DEFINED BLAS_VENDOR OR BLAS_VENDOR STREQUAL "All"))
```

Reply:
Don't ask me why, but [it really is `BLA_VENDOR`](https://cmake.org/cmake/help/latest/module/FindBLAS.html#input-variables) :sweat_smile:

Reply:
😭